modules/core/src/main/scala/ch/epfl/bluebrain/nexus/admin/core/resources/Resources.scala

```diff
@@ -153,14 +150,13 @@ class Resources[F[_], A: IdResolvable: TypeFilterExpr](agg: Agg[F], sparqlClient
     *
     * @param id     the identifier of the resource
     * @param rev    the revision attempted to be fetched
-    * @param persId the unique persistence ID of the resource
     * @return an optional [[Resource]] instance wrapped in the
     *         abstract ''F[_]'' type if successful, or a [[ch.epfl.bluebrain.nexus.admin.core.Fault]] wrapped within
     *         ''F[_]'' otherwise
     */
-  def fetch(id: A, rev: Long)(persId: String)(implicit @silent perms: HasReadProjects): F[Option[Resource[A]]] =
-    stateAt(persId, rev).map {
-      case c: Current if c.rev == rev => Some(Resource(id, c.rev, c.value, c.deprecated))
+  def fetch(id: A, rev: Long): F[Option[Resource[A]]] =
```

Comment:
you need an evidence that you can call that method, otherwise anybody can.
`(implicit @silent perms: HasReadProjects)`